locator/token_metadata.cc

```diff
@@ -310,6 +312,22 @@ class token_metadata_impl final {
         tlogger.debug("ring_version={}", _ring_version);
     }
 
+    token_metadata::version_t get_version() const {
+        return _version;
+    }
+    void set_version(token_metadata::version_t version) {
+        if (version <= 0) {
+            on_internal_error(tlogger,
+                format("token_metadata_impl::set_version: invalid new version {}", version));
+        }
+        if (version < _version) {
+            on_internal_error(tlogger,
+                format("token_metadata_impl::set_version: new version can't be smaller than the previous one, "
+                       "new version {}, previous version {}", version, _version));
+        }
+        _version = version;
+    }
```

Comment:
Q (not directly related):

How are we protected from this node losing leadership, some other node advancing the topology, and then this node regaining leadership and committing a now-outdated _version?